src/ops/custom.rs

```diff
@@ -222,7 +222,10 @@ pub fn resolve_extension_ops(
     for n in h.nodes() {
         if let OpType::LeafOp(LeafOp::CustomOp(op)) = h.get_optype(n) {
             if let ExternalOp::Opaque(opaque) = op.as_ref() {
-                if let Some(r) = extension_registry.get(&opaque.extension) {
+                if let Some(r) = extension_registry.get({
+                    let k: &str = &opaque.extension;
+                    k
+                }) {
```

Comment:
Yeah, this is annoying. I tried just `&opaque.extension` and the typechecker complains:
```
error[E0277]: the trait bound `smol_str::SmolStr: Borrow<IdentList>` is not satisfied
   --> src/ops/custom.rs:225:57
    |
225 | ...extension_registry.get(&opaque.extension) {
    |                       --- ^^^^^^^^^^^^^^^^^ the trait `Borrow<IdentList>` is not implemented for `smol_str::SmolStr`
    |                       |
    |                       required by a bound introduced by this call
    |
    = help: the trait `Borrow<str>` is implemented for `smol_str::SmolStr`
```
so that `: &str` forces the `&` to use the `Borrow<str>` of IdentList to get a `&str` and then use the `Borrow<str>` of SmolStr (the keys in the map) that the message suggests. Do you know of a better way?

Reply:
Your `as &str` works 👍 
Another solution would have been `&*opaque.extensions` (borrow and then get the string slice)

Reply:
Yes, I thought there might be something like that. That was the kinda obvious "thing like that" and indeed it works...I've done that here but kept the tests with the explicit `as &str`.